odoo/fields.py

```diff
@@ -1195,9 +1195,21 @@ def __get__(self, record, owner):
                 value = env.cache.get(record, self)
 
             elif self.store and record._origin and not (self.compute and self.readonly):
-                # new record with origin: fetch from origin
-                value = self.convert_to_cache(record._origin[self.name], record, validate=False)
-                env.cache.set(record, self, value)
+                # new record with origin: fetch from origin, and assign the
+                # records to prefetch in cache (which is necessary for
+                # relational fields to "map" prefetching ids to their value)
+                recs = record._in_cache_without(self)
+                try:
+                    for rec in recs:
+                        if rec._origin:
+                            value = self.convert_to_cache(rec._origin[self.name], rec, validate=False)
+                            env.cache.set(rec, self, value)
+                    value = env.cache.get(record, self)
+                except Exception:
```

Comment:
I am ok with the fallback but only for `(AccessError, MissingError)`. I don't understand why this fallback should manage more cases than the one inside the `elif self.compute:` ? :thinking: